### DISCUSSION TITLE
Remove dynamics from simulation

### DISCUSSION THREAD

**Discussion by aleada23** [google-deepmind/mujoco#1533] [General]
Good morning. I'm a student and I'm using MuJoCo for a project. I know that dynamic is an important feature for this engine, but I need to remove it from my environment. 
I need to do this since I want to implement a reinforcement learning algorithm in my project and I need to speed up simulations. I defined a robot in simulation and I'm performing tasks like "collision avoidance" and "inverse kinematics". So I can read of dynamics from my simulations keeping only kinematics, supposing that the robot is an ideal one without compromising the learning part of the project. I looked for a solution and the simplest way seems to be to set the parameters to zero. However I don't think that this way will really help me in making the computation lighter.
Any suggestion about this?
Thank you for your time

**Comment by yuvaltassa** [comment]
Your question is very difficult to parse. I think that you want to just not call `mj_step` but only `mj_kinematics` or `mj_fwdPosition`? Or set all velocities to zero? Not entirely sure what you're asking.

**Comment by aleada23** [comment]
Sorry, let me try to be clearer. 
I've created an environment, also using Robosuite, where a robot has to perform some tasks. Now, I need to remove the dynamics from the environment to make computation lighter for implementing a new learning component in my project. The dynamics currently does not constitute a fundamental part of my simulation during the learning phase, so I can do without it. I am only interested in keeping some Mujoco functionalities active, such as the rangefinder sensors, or obtain Jacobian for inverse kinematics for robot control. Therefore, I would like to remove its effect. For instance, if I were to zero out masses and frictions, I would remove a significant portion of the dynamics from the simulation, but I believe the calculations would still be performed, keeping the simulation slow as it is now. So, is it possible to remove the dynamics? I would like to avoid any calculations related to it during a simulation, as it isn't necessary in this new phase.
Thank you
